### ISSUE TITLE
Overridde `DEM.crs` with `DEM.ccrs` to respect 3D CRSs

### ISSUE THREAD

**Issue by rhugonnet** [GlacioHack/xdem#370]
For instance useful if saving to file. Not a priority, but could be a good feature in the long term when 3D CRSs get more popular.

We'd have to test thoroughly if all types of 3D CRS can be processed (I don't think saving will work for grids, for example, we'd have to raise a `UserWarning` then).

> Regarding your suggestion to override crs with vcrs, do you mean to override crs with ccrs, or am I missing something? I'm still a bit new to vcrs so it might be a dumb question!

Yes, exactly, overriding it with `ccrs` so that it's always 3D! (then I guess we could remove `ccrs` and have it just be `crs`).

@adehecq: Curious about your feedback on this PR as well! 
FYI: It's a problem also of interest to the ICESat-2 SlideRule team that wants to integrate vertical transforms in their cloud processing, so we're discussing it a bit there too in parallel.

_Originally posted by @rhugonnet in https://github.com/GlacioHack/xdem/issues/350#issuecomment-1522538814_
